src/main/java/com/bootcamp/topic3/service/ShoppingCartServiceImp.java

```diff
@@ -0,0 +1,59 @@
+package com.bootcamp.topic3.service;
+
+import java.util.ArrayList;
+
+import com.bootcamp.topic3.cart.CartItem;
+import com.bootcamp.topic3.cart.ShoppingCart;
+
+public class ShoppingCartServiceImp implements ShoppingCartService {
+	
+	private ArrayList<CartItem> items;
+	
+	
+	public ShoppingCartServiceImp() {
+		items = new ArrayList<CartItem>();
+	}
+
+	/**
+	 * agrega un item al carro
+	 */
+	public void addCartItem(CartItem item) {
+		items.add(item);
+	}
+
+	/**
+	 * borra un item del carro
```

Comment:
The javadoc should be on the interface, the javadoc in the implemetation will be copied from there.

> The interface should have all the information about the contract, basically what the method does, the description of parameters, return values and so on.

> Unless there's some extra information that isn't clear from the interface description (there rarely is), the implementation documentation should then simply link to the interface method.

Also the implementation's javadoc can be override
https://dzone.com/articles/inheriting-javadoc-method-comments